tests/cli/main.cpp

```diff
@@ -0,0 +1,270 @@
+      account_object nathan_acct_before_upgrade = wapiptr->get_account("nathan");
+
+      // upgrade nathan
+      BOOST_TEST_MESSAGE("Upgrading Nathan to LTM");
+      signed_transaction upgrade_tx = wapiptr->upgrade_account("nathan", true);
+      account_object nathan_acct_after_upgrade = wapiptr->get_account("nathan");
+
+      // verify that the upgrade was successful
+      BOOST_CHECK_PREDICATE( std::not_equal_to<uint32_t>(), (nathan_acct_before_upgrade.membership_expiration_date.sec_since_epoch())(nathan_acct_after_upgrade.membership_expiration_date.sec_since_epoch()) );
+      BOOST_CHECK(nathan_acct_after_upgrade.is_lifetime_member());
+
+      // create a new account
+      graphene::wallet::brain_key_info bki = wapiptr->suggest_brain_key();
+      BOOST_CHECK(!bki.brain_priv_key.empty());
+      signed_transaction create_acct_tx = wapiptr->create_account_with_brain_key(bki.brain_priv_key, "jmjatlanta", "nathan", "nathan", true);
+      // save the private key for this new account in the wallet file
+   	  BOOST_CHECK(wapiptr->import_key("jmjatlanta", bki.wif_priv_key));
+      wapiptr->save_wallet_file(wallet_filename.str());
+
+      // attempt to give jmjatlanta some bitsahres
+      BOOST_TEST_MESSAGE("Transferring bitshares from Nathan to jmjatlanta");
+      signed_transaction transfer_tx = wapiptr->transfer("nathan", "jmjatlanta", "10000", "BTS", "Here are some BTS for your new account", true);
+
+      // get the details for init1
+      witness_object init1_obj = wapiptr->get_witness("init1");
+      int init1_start_votes = init1_obj.total_votes;
+      // Vote for a witness
+      signed_transaction vote_witness1_tx = wapiptr->vote_for_witness("jmjatlanta", "init1", true, true);
+
+      // wait for a maintenance interval
+      // NOTE: For this to work consistently, your maintenance interval must be less than 20 seconds
+      fc::usleep(fc::seconds(20));
+      // send a block to trigger maintenance interval
+      BOOST_CHECK(generate_block(*app1.get()));
+
+      // Verify that the vote is there
+      init1_obj = wapiptr->get_witness("init1");
+      witness_object init2_obj = wapiptr->get_witness("init2");
+      int init1_middle_votes = init1_obj.total_votes;
+      BOOST_CHECK(init1_middle_votes > init1_start_votes);
+
+      // Vote for a 2nd witness
+      int init2_start_votes = init2_obj.total_votes;
+      signed_transaction vote_witness2_tx = wapiptr->vote_for_witness("jmjatlanta", "init2", true, true);
+
+      // send another block to trigger maintenance interval
+      fc::usleep(fc::seconds(20));
+      BOOST_CHECK(generate_block(*app1.get()));
+
+      // Verify that both the first vote and the 2nd are there
+      init2_obj = wapiptr->get_witness("init2");
+      init1_obj = wapiptr->get_witness("init1");
+
+      int init2_middle_votes = init2_obj.total_votes;
+      BOOST_CHECK(init2_middle_votes > init2_start_votes);
```

Comment:
this test is failing.

Reply:
I added a check to make sure the maintenance interval is set to a low value for this test. Is there a way to adjust the maintenance interval at run time?

Reply:
this can be done by another way, check for example:

https://github.com/bitshares/bitshares-core/blob/master/tests/tests/swan_tests.cpp#L114

in there, block production advances until a hardfork time, you can play with that in order to advance the chain to the time you need. it will be something like:

vote for witness
advance until next maintenance
check for vote

looking good :)

Reply:
Take a look at https://github.com/jmjatlanta/bitshares-core/blob/b6a6dc3a1f3b3af1f8b3e65f2392863fb306670c/tests/cli/main.cpp#L113

It is a modified version of what is in database_fixture. No more playing with the default maintenance interval